File: js/leetcode/jump-game.js
/**
 * 数组的数值表示最长的跳跃距离
 * 数组个数 >= 1
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let maxDist = nums[0]; // 起点
    for(let i = 1; i <= maxDist && i <= nums.length - 1; ++i) {
        maxDist = Math.max(i + nums[i], maxDist);
    }
    return maxDist >= nums.length-1;
};

File: java/src/leetcode/JumpGame.java
package leetcode;
import java.lang.Math;

public class JumpGame {

    class Solution {
        public boolean canJump(int[] nums) {
            int max = 0;
            for (int i = 0; i < nums.length && max >= i; i++) {
                max = Math.max(i + nums[i] , max);
            }
            return max >= nums.length - 1;
        }
    }

    public static void main(String[] args) {
        JumpGame.Solution solution = new JumpGame().new Solution();
        solution.canJump(new int[] {2,3,1,1,4});
    }
}



